WalletWasabi/Extensions/TaskExtensions.cs

```diff
@@ -16,22 +16,67 @@ public static async Task<T> WithAwaitCancellationAsync<T>(this Task<T> me, Cance
 						s => ((TaskCompletionSource<bool>)s).TrySetResult(true), tcs))
 			{
 				// If the task waited on is the cancellation token...
-				if (me != await Task.WhenAny(me, tcs.Task))
+				if (me != await Task.WhenAny(me, tcs.Task).ConfigureAwait(false))
 				{
-					if (waitForGracefulTermination <= 0)
+					if (waitForGracefulTerminationMilliseconds <= 0)
 					{
 						throw new OperationCanceledException(cancel);
 					}
 					else
 					{
-						using var cts = new CancellationTokenSource(waitForGracefulTermination);
-						return await me.WithAwaitCancellationAsync(cts.Token);
+						using var cts = new CancellationTokenSource(waitForGracefulTerminationMilliseconds);
+						return await me.WithAwaitCancellationAsync(cts.Token).ConfigureAwait(false);
 					}
 				}
 			}
 
 			// Wait for one or the other to complete.
-			return await me;
+			return await me.ConfigureAwait(false);
 		}
+
+		/// <summary>
+		/// UNSAFE! You are cancelling the wait on the callback of the original Task, not cancelling the operation itself.
+		/// https://stackoverflow.com/questions/14524209/what-is-the-correct-way-to-cancel-an-async-operation-that-doesnt-accept-a-cance/14524565#14524565
+		/// </summary>
+		public static async Task WithAwaitCancellationAsync(this Task me, CancellationToken cancel, int waitForGracefulTerminationMilliseconds = 0)
+		{
+			// The tasck completion source.
```

Comment:
```suggestion
			// The task completion source.
```
I know this was a copy paste so maybe the original code should be corrected also.

Reply:
Done.